akka-docs/src/test/scala/docs/persistence/PersistencePluginDocSpec.scala

```diff
@@ -47,6 +47,16 @@ object PersistencePluginDocSpec {
       //#native-config
       akka.persistence.journal.leveldb.native = off
       //#native-config
+
+      //#compaction-intervals-config
+      # Number of deleted messages per persistence id that will trigger journal compaction
+      akka.persistence.journal.leveldb.compaction-intervals {
+        # persistence-id-1 = interval-1
+        # persistence-id-2 = interval-2
+        # ...
+        # persistence-id-N = interval-N
```

Comment:
I thought of the possibility of wildcards as well, but that is a more complex issue than what it initially seems. If you check the signature and documentation of the **compactRange** method of the LevelDB API, you will see that the start-end parameters are byte arrays, which are calculated by the LeveldbStore using the Key utility object (akka api). The Key object in turn uses the persistence ids to calculate the most significant bytes of these arrays, thus implying that the messages of each distinct persistence-id are stored in a continuous range, which is mutually exclusive to other ranges (obviously for performance reasons). This however is the very reason that a wildcard-based compaction could problematic: it would probably lead to performing compaction across multiple ranges, which would effectively try to compact undeleted messages as well. This makes no sense as compaction does nothing when it comes to messages without tombstones, but I have the feeling that due to the much wider range of the scan, it would come at a significant performance cost. I think this may be what is implied by the documentation string of the compactRange method found [here](https://github.com/google/leveldb/blob/master/include/leveldb/db.h) which says: 

> This operation should typically only be invoked by users who understand the underlying implementation.

Given the thoughts above, I kind of dropped the wildcard though for now.